packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -1130,6 +1130,40 @@ void main() {
     }
   });
 
+  testWidgets('Enable filtering with custom filter callback that filter text case sensitive', (WidgetTester tester) async {
+    final ThemeData themeData = ThemeData();
+    final TextEditingController controller = TextEditingController();
+    addTearDown(controller.dispose);
+
+    await tester.pumpWidget(MaterialApp(
+      theme: themeData,
+      home: Scaffold(
+        body: DropdownMenu<TestMenu>(
+          requestFocusOnTap: true,
+          enableFilter: true,
+          filterCallback: (List<DropdownMenuEntry<TestMenu>> entries, String filter) {
+            return entries.where((DropdownMenuEntry<TestMenu> element) => element.label.contains(filter)).toList();
+          },
+          dropdownMenuEntries: menuChildren,
+          controller: controller,
+        ),
+      ),
+    ));
+
+    // Open the menu
+    await tester.tap(find.byType(DropdownMenu<TestMenu>));
+    await tester.pump();
+
+    await tester.enterText(find.byType(TextField).first, 'item');
+    expect(controller.text, 'item');
+    await tester.pumpAndSettle();
+    for (final TestMenu menu in TestMenu.values) {
+      // All items should appear only once. This meaning that search returned
+      // nothing
+      expect(find.widgetWithText(MenuItemButton, menu.label), findsOneWidget);
```

Comment:
Can you please also update this test? This unit test doesn't really do the case-sensitive check as we mentioned before. Here's an updated version that you can refer: https://github.com/flutter/flutter/pull/143939#discussion_r1617878025

Reply:
@QuncCccccc I missed that comment, sorry 🙈 

We should be fine now